tests/api.cpp

```diff
@@ -59,133 +70,69 @@ int main( int argc, char** argv )
    try {
```

Comment:
Can you transform this into a proper unit test that runs as part of `all_tests` instead of having an executable of its own?